repositories/strategies/buildTestQuestionnaire.js

```diff
@@ -0,0 +1,254 @@
+const QuestionnaireRepository = require("../QuestionnaireRepository");
+const SectionRepository = require("../SectionRepository");
+const PageRepository = require("../PageRepository");
+const AnswerRepository = require("../AnswerRepository");
+const OptionRepository = require("../OptionRepository");
+const ValidationRepository = require("../ValidationRepository");
+const { getValidationEntity } = require("./validationStrategy");
+const { validationRuleMap } = require("../../utils/defaultAnswerValidations");
+const MetadataRepository = require("../MetadataRepository");
+
+const replacePiping = (field, references) => {
+  if (field.indexOf("<span") === -1) {
+    return field;
+  }
+
+  const id = /data-id="(.+?)"/.exec(field)[1];
+  const pipeType = /data-piped="(.+?)"/.exec(field)[1];
+
+  const newId = references[pipeType][id];
+
+  return field.replace(`data-id="${id}"`, `data-id="${newId}"`);
+};
+
+const buildValidations = async (validationConfigs = {}, answer) => {
+  const validationEntityType = getValidationEntity(answer.type);
+  const validationTypes = validationRuleMap[validationEntityType] || [];
+
+  let validations = {};
+  for (let i = 0; i < validationTypes.length; ++i) {
```

Comment:
Unfortunately we can't as this would trigger all the duplications to run simultaneously.

We could create a chain of Promises instead but this is nicer. IMO